policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,246 @@
+use crate::{
+    http_route::{ParentReference, RouteBinding},
+    resource_id::ResourceId,
+};
+use ahash::{AHashMap as HashMap, AHashSet as HashSet};
+use chrono::offset::Utc;
+use linkerd_policy_controller_k8s_api::{self as k8s, gateway, ResourceExt};
+use parking_lot::RwLock;
+use std::{collections::hash_map::Entry, sync::Arc};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+const POLICY_API_GROUP: &str = "policy.linkerd.io";
+const POLICY_API_VERSION: &str = "policy.linkerd.io/v1alpha1";
+const STATUS_CONTROLLER_NAME: &str = "policy.linkerd.io/status-controller";
+
+pub type SharedIndex = Arc<RwLock<Index>>;
+
+pub struct Controller {
+    client: k8s::Client,
+    index: SharedIndex,
+    updates: UnboundedReceiver<Update>,
+}
+
+pub struct Index {
```

Comment:
Is it useful have separate structs for Controller and Index like this?  Having the sender and receiver on different structs gives the impression that the channel is being used to send data from one place to another, but that's not really what's happening (Index is effectively a substruct of Controller).  In actuality we're just using the channel as a queue so that we can process updates asynchronously.  I think that comes across more clearly if we just use a single flat struct which holds both the tx and rx.

Reply:
There have been some changes since this comment, but I think it still makes sense to keep them separate. `Controller` no longer holds on to an `Index` and just processes update; all updates are created synchronously as part of `Index`. The separation makes testing much nicer as well.